Documentation/ProvenHeaders.md

```diff
@@ -0,0 +1,229 @@
+# ProvenHeaders specifications
+## Table of contents
+
+- [Problem overview](#problem-overview)
+
+- [Solution abstract overview](#solution-abstract-overview)
+
+- [Basic constructs](#basic-constructs)
+
+- [Structure of proven header](#structure-of-proven-header)
+
+- [Assumptions](#assumptions)
+
+- [Best chain selection](#best-chain-selection)
+
+- [Changes required for the C# node](#changes-required-for-the-c-node)
+
+  ​
+
+
+## Problem overview
+
+Syncing with headers is used for quick blockchain download. Node can ask its peers for headers using *getheaders* message and asked peer will send up to 2000 headers per *headers* message. When the headers are received they are used in order to ask peers for blocks that are represented by those headers. This protocol is new and used on bitcoin network because it has some advantages in terms of resources usage over using *inv* messages.
```

Comment:
```suggestion
Syncing with headers is used for quick blockchain downloads. A node can ask its peers for headers using *getheaders* message and the asked peer can send up to 2000 headers per *headers* message. When the headers are received, they are used in order to ask peers for blocks those headers represent. This protocol is new and used on bitcoin network because it has some advantages in terms of resources usage over using *inv* messages.
```